### PR TITLE
#149 add docs on building locally

### PR DESCRIPTION
**Pull request by hgword**

### CODE REVIEW

File: BUILDING.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
+## How to Build Squash
+
+### Pre-requirement
```
Comment (mitchdraft) [inline BUILDING.md]:
```suggestion
### Requirements
```

File: changelog/v0.5.6/add-building-doc.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+changelog:
+- type: FIX
+  description: Add document file that how to build squash
```
[the remark this answers is not in the document]
Comment (hgword) [reply changelog/v0.5.6/add-building-doc.yaml]: @mitchdraft 
I apply your code comment 436d0713328ce0caa042388299cad2e5be8f0818

File: BUILDING.md (hunk 1 above)
Comment (hgword) [reply BUILDING.md]: @mitchdraft 
I apply your code review 35a5692afdc43914a6f601716e29a01b2a292975

File: BUILDING.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    # To see the progress, check the pkg directory size:
+    # cd $GOPATH/pkg ; du -sh
+    dep ensure
+  ```
+
+### Now you can build Squash
+  ```bash
+    make build
+  ```
+
```
[the remark this answers is not in the document]
Comment (hgword) [reply BUILDING.md]: @mitchdraft 
I apply your code review 35a5692afdc43914a6f601716e29a01b2a292975

File: BUILDING.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    # Download the source by using go get
+    go get github.com/solo-io/squash
+  
+    # or by using git clone
+    # mkdir -p $GOPATH/src/github.com/solo-io
+    # cd $GOPATH/src/github.com/solo-io
+    # git clone https://github.com/solo-io/squash.git
+    
+    cd $GOPATH/src/github.com/solo-io/squash
+    git checkout -b v0.5.5 tags/v0.5.5
```
[the remark this answers is not in the document]
Comment (hgword) [reply BUILDING.md]: @mitchdraft 
I apply your code review 35a5692afdc43914a6f601716e29a01b2a292975

File: BUILDING.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  ```bash
+    # Set env variables and add Go to the PATH in .bashrc / .zshrc file.
+    export GOPATH=$HOME/golang
+    export GOROOT=/usr/lib/go
+    export PATH=$PATH:$GOPATH/bin:$GOROOT/bin
+  ```
+
+* make $GOPATH directory
+  ```bash
+    mkdir -p $GOAPTH/src
```
[the remark this answers is not in the document]
Comment (hgword) [reply BUILDING.md]: @mitchdraft 
I apply your code review 35a5692afdc43914a6f601716e29a01b2a292975

File: BUILDING.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    cd $GOPATH/src/github.com/solo-io/squash
+    git checkout -b v0.5.5 tags/v0.5.5
+  ```
+
+* install the dependencies
+  ```bash
+    # This job may take some time.
+    # To see the progress, check the pkg directory size:
+    # cd $GOPATH/pkg ; du -sh
+    dep ensure
```
[the remark this answers is not in the document]
Comment (hgword) [reply BUILDING.md]: @mitchdraft 
I apply your code review 35a5692afdc43914a6f601716e29a01b2a292975

**Pull request merged**